TigranKirakosov/index.js

```diff
@@ -0,0 +1,253 @@
+var UIController = (function() {
+  var DOMStrings;
+  DOMStrings = {
+    taskInput: '.input-text',
+    deadlineInput: '#deadline',
+    submitBtn: '.submit-btn',
+    filterInput: '#filter',
+    filterBtn: '.filter-btn',
+    uncheckedBtn: 'far fa-circle',
+    checkedBtn: 'far fa-check-circle',
+    deleteBtn: 'fas fa-backspace',
+    taskList: '.collection',
+    clearBtn: '.clear-btn'
+  };
+
+  return {
+    getInput: function() {
+      return {
+        text: document.querySelector(DOMStrings.taskInput).value,
+        deadline: document.querySelector(DOMStrings.deadlineInput).value,
+        filterValue: document.querySelector(DOMStrings.filterInput).value
+      };
+    },
+    getDOMStrings: function() {
+      return DOMStrings;
+    },
+    deleteListItem: function(selectorID) {
+      var element;
+      element = document.querySelector(`#${selectorID}`);
+      element.parentNode.removeChild(element);
+    },
+    toggleTask: function(icon) {
+      if (icon.className === DOMStrings.checkedBtn) {
+        icon.className = DOMStrings.uncheckedBtn;
+      } else if (icon.className === DOMStrings.uncheckedBtn) {
+        icon.className = DOMStrings.checkedBtn;
+      }
+    },
+    filterTasks: function(tasks) {

```

Comment:
why UI controller should know how to filter data? As I can see in your to do list `UIController` - it's classic view component. All what should do this controller - correctly display data to user. Filtering of the data should do `controller` (line 145) component and just put array of filtered tasks to UI controller